src/many-abci/src/module.rs

```diff
@@ -102,6 +115,72 @@ fn _tm_query_from_many_filter(
     Ok(query)
 }
 
+fn transform_list_args(
+    blockchain::ListArgs {
+        count,
+        order,
+        filter,
+    }: blockchain::ListArgs,
+) -> Result<(u64, Order, Query), ManyError> {
+    filter
+        .map_or(
+            Ok(DEFAULT_BLOCK_LIST_QUERY.clone()),
+            _tm_query_from_many_filter,
+        )
+        .map(|filter| {
+            (
+                count.map_or(MAXIMUM_BLOCK_COUNT, |c| {
+                    std::cmp::min(c, MAXIMUM_BLOCK_COUNT)
+                }),
+                order.map_or(tendermint_rpc::Order::Ascending, _tm_order_from_many_order),
+                filter,
+            )
+        })
+}
+
+fn create_pagination(count: u64) -> Result<(u32, u8), ManyError> {
+    // We can get maximum u8::MAX blocks per page and a maximum of u32::MAX pages
+    // Find the correct number of pages and count
+    let maximum_8_bit_integer: u64 = u8::MAX.into();
+    Ok((
+        num_integer::div_ceil(count, maximum_8_bit_integer)
+            .try_into()
+            .map_err(|_| ManyError::unknown("Unable to cast u64 to u32"))?,
+        std::cmp::max(count, maximum_8_bit_integer)
+            .try_into()
+            .map_err(|_| ManyError::unknown("Unable to cast u64 to u8"))?,
+    ))
+}
+
+fn tx_results<C: Client + Sync>(
+    client: &C,
+    count: u64,
+    order: Order,
+    query: Query,
+) -> Result<Vec<(TransactionIdentifier, Vec<u8>)>, ManyError> {
+    use futures_util::future::TryFutureExt;
+    let (pages, count) = create_pagination(count)?;
+    block_on(
+        client
+            .tx_search(query, true, pages, count, order)
+            .map_err(|_| ManyError::unknown("Transaction search query returned an error"))
+            .and_then(|response| async move {
+                Ok(response
+                    .txs
+                    .iter()
+                    .map(|tx| {
+                        (
+                            TransactionIdentifier {
+                                hash: hash_tx(&tx.tx),
```

Comment:
Placing here as a duplicate of a response to a previous code review comment:

`hash_tx(&tx.tx)` and `tx.hash` are not of the same type. `hash_tx(&tx.tx)` is a `Vec<u8>`, while `tx.hash` is a `tendermint_rpc::abci::transaction::Hash`.

Reply:
Can you convert (even just `.into()`) `tendermint_rpc::abci::transaction::Hash` into a `Vec<u8>`? This is costing computing time for no reason, you already have the hash, right?

Reply:
Seems like you can just `.into()`: https://github.com/informalsystems/tendermint-rs/blob/main/tendermint/src/hash.rs#L49-L56. What am I missing?

Reply:
@hansl Yes, I can do this.  Thanks for sharing the source code for that type.

Reply:
@hansl I tried `into`, but it didn't work: 

```
the trait bound `Vec<u8>: std::convert::From<tendermint_rpc::abci::transaction::Hash>` is not satisfied
```

Reply:
I'd investigate that further, seems like there's something missing or wrong. Anyway, for another PR.